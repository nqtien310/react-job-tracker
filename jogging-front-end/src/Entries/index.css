table td {
  width: 180px;
}

table td.actions {
  width: 150px;
}

table td.date {
  width: 400px;
  font-size: 14px;
  font-weight: bold;
}


table td input.form-control{
  width: 100px;
}

.field-container {
  width: 130px;
  margin-right: 50px;
  display: inline-block;
}

.field-container.time {
  width: 220px;
}

.field-container.time input{
  display:inline-block;
  width: 130px;
  margin-right: 10px;
}

.date-filters {
  margin-bottom: 20px
}

.date-filters > div {
  display: inline-block;
}

.date-filter {
  width: 150px;
  margin-left: 30px;
}
