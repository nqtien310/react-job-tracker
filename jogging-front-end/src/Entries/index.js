import React from 'react';
import { connect } from 'react-redux'
import { deleteEntry } from '../epics/deleteEntryEpic'
import List from './list'
import Form from './form'
import './index.css'

class Entries extends React.Component{
  constructor(props) {
    super(props)
  }

  render(){
    return (
      <div>
        <List userId={this.props.userId} entries={this.props.entries}/>
        <Form/>
      </div>
    )
  }
}

function mapStateToProps(state){
  return {
    entries: state.entries
  }
}
export default connect(mapStateToProps, {
  deleteEntry
})(Entries)
