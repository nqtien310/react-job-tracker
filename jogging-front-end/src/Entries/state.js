import Rx from 'rxjs/Rx'
import api from '../api'
import { combineReducers } from 'redux'
import { push } from 'react-router-redux'
import setErrorMessage from '../actions/setErrorMessage.js'
import {reset} from 'redux-form'
//constants
export const ENTRY_FORM_CREATE = 'ENTRY_FORM_CREATE'
export const ENTRY_FORM_EDIT   = 'ENTRY_FORM_EDIT'
export const ENTRY_FORM_HIDE   = 'ENTRY_FORM_HIDE'
export const ENTRY_DELETE      = 'ENTRY_DELETE'
export const ENTRY_CREATE      = 'ENTRY_CREATE'
export const ENTRY_UPDATE      = 'ENTRY_UPDATE'
export const ENTRIES_FETCH     = 'ENTRIES_FETCH'
export const ENTRIES_FETCHED   = 'ENTRIES_FETCHED'

//actions
export const deleteEntry = (userId, entryId) => {
  return {
    type: ENTRY_DELETE,
    payload:{
      userId: userId,
      id: entryId
    }
  }
}

export const fetchEntries = (userId) => ({type: ENTRIES_FETCH, payload: userId})
export const showCreateForm = () => ({type: ENTRY_FORM_CREATE})
export const showEditForm = (entry) => ({type: ENTRY_FORM_EDIT, payload: entry})
export const hideForm = (entry) => ({type: ENTRY_FORM_HIDE})
export const createEntry = (userId, params) => {
  return {
    type: ENTRY_CREATE,
    payload: {
      userId: userId,
      params: params
    }
  }
}

export const updateEntry = (userId, id, params) => {
  return {
    type: ENTRY_UPDATE,
    payload: {
      userId: userId,
      id: id,
      params: params
    }
  }
}

export const fetchedEntries = (response) => {
  return {
    type: ENTRIES_FETCHED,
    payload: response.data
  }
}

//reducers
function listReducer(state=[], action){
  switch(action.type){
    case ENTRIES_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

function formReducer(state={}, action){
  switch(action.type){
    case ENTRY_FORM_CREATE: {
      return {action: 'create'}
    }case ENTRY_FORM_EDIT: {
      return {action: 'edit', editing: action.payload}
    }case ENTRY_FORM_HIDE: {
      return {}
    }default: {
      return state
    }
  }
}

export const entryReducer = combineReducers({
  list: listReducer,
  form: formReducer
})

//Epics

export const deleteEntryEpic = (action$) => {
  return action$.ofType(ENTRY_DELETE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.delete(`users/${action.payload.userId}/entries/${action.payload.id}`)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const updateEntryEpic = (action$) => {
  return action$.ofType(ENTRY_UPDATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.put(`users/${action.payload.userId}/entries/${action.payload.id}`, action.payload.params)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const fetchEntriesEpic = (action$) => {
  return action$.ofType(ENTRIES_FETCH).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.get(`users/${action.payload}/entries`)
    ).map(response => fetchedEntries(response))
  })
}

export const createEntryEpic = (action$) => {
  return action$.ofType(ENTRY_CREATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.post(`users/${action.payload.userId}/entries`, action.payload.params)
    ).flatMap(() => [fetchEntries(action.payload.userId), hideForm()]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}


//https://github.com/redux-observable/redux-observable/tree/master/examples/navigation/epics
