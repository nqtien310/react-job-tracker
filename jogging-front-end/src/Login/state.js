import { fetchMyUser } from '../epics/fetchMyUserEpic'
import Auth from '../auth'
import 'rxjs/add/operator/switchMap';
import Rx from 'rxjs/Rx'
import api from '../api'
import setErrorMessage from '../actions/setErrorMessage.js'

export const LOGIN  = 'LOGIN'
export const SETTOKEN = 'SETTOKEN'

export const submitLogin = params => ({type: LOGIN, payload: params})
export const setToken    = params => ({type: SETTOKEN, payload: params})


export const loginEpic = (action$) => {
  return action$.ofType(LOGIN).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.post('login', {user: action.payload})
    ).flatMap(response =>
        [setToken(response), fetchMyUser()]
      ).catch(error => setErrorMessage(error))
  })
}

