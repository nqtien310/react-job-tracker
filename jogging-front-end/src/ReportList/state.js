import Auth from '../auth'
import api from '../api'
import Rx from 'rxjs/Rx'
import setErrorMessage from '../actions/setErrorMessage.js'
import { combineReducers } from 'redux'

export const REPORTS_FETCH = 'REPORTS_FETCH'
export const REPORTS_FETCHED = 'REPORTS_FETCHED'

export const fetchReports = (userId) => {
  return {
    type: REPORTS_FETCH,
    payload:{
      userId: userId
    }
  }
}

export const fetchedReports = (response) => {
  return {
    type: REPORTS_FETCHED,
    payload: response.data
  }
}

const listReducer = (state=[], action) => {
  switch(action.type){
    case REPORTS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const reportReducer = combineReducers({
  list: listReducer
})

export const fetchReportsEpic = (action$) => {
  return action$.ofType(REPORTS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(
      api.get(`/users/${action.payload.userId}/weekly_summaries`)
    ).flatMap(response => [fetchedReports(response)])
  })
}
