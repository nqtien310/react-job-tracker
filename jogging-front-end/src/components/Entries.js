import React from 'react';
import { connect } from 'react-redux'
import { deleteEntry } from '../epics/deleteEntryEpic'

class Entries extends React.Component{
  constructor(props) {
    super(props)
  }

  onDelete(id){
    this.props.deleteEntry(this.props.userId, id)
  }

  renderEntries() {
    return this.props.entries.map(entry => {
      return(
        <tr key={entry.id}>
          <td>
            {entry.distance_in_metre} m
          </td>
          <td>
            {entry.speed} m/s
          </td>
          <td>
            {entry.time_in_second} s
          </td>
          <td>
            {entry.formatted_date}
          </td>
          <td>
            <a onClick={()=> this.onDelete(entry.id)} className="btn btn-danger">Delete</a>
          </td>
        </tr>
      )
    })
  }

  render(){
    return (
      <table className="table">
        <thead>
          <tr>
            <th>Distance</th>
            <th>Speed</th>
            <th>Time In Second</th>
            <th>Date</th>
            <th>Action</th>
          </tr>
        </thead>

        <tbody>
          {this.renderEntries()}
        </tbody>
      </table>
    )
  }
}

function mapStateToProps(state){
  return {
    entries: state.entries
  }
}
export default connect(mapStateToProps, {
  deleteEntry
})(Entries)
