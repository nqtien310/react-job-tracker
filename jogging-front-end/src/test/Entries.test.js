import React from 'react';
import renderer from 'react-test-renderer'
import LoginForm from '../Login'
import RenderComponent from './RenderComponent'
import { store } from './RenderComponent'
import Entries from '../Entries'
import { stubApiSuccess } from './ApiStubs'
import { createEntry, updateEntry, deleteEntry } from '../Entries/state'

it('Listing (No Entries)', (done) => {
  stubApiSuccess("get", [])

  const component = renderer.create(
    <RenderComponent>
      <Entries reportLink="/report" myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout(() => {
    expect(component).toMatchSnapshot()
    done()
  },50)
});

it('Listing (With entries)', (done) => {
  let data = [
    {"id":42,"time_in_second":100,"distance_in_metre":100.0,"formatted_date":"2017-09-03","speed":1.0,"date":"2017-09-03"},
    {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
  ]

  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <Entries reportLink="/report" myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout(() => {
    expect(component).toMatchSnapshot()
    done()
  },50)
});

it('Delete entry', (done) => {
  let data = [
    {"id":42,"time_in_second":100,"distance_in_metre":100.0,"formatted_date":"2017-09-03","speed":1.0,"date":"2017-09-03"},
    {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
  ]

  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <Entries reportLink="/report" myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout(() => {
    let dataAfterDelete = [
      {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
    ]

    expect(component).toMatchSnapshot()
    stubApiSuccess("get", dataAfterDelete)
    stubApiSuccess("delete", null)
    store.dispatch(deleteEntry(1,42))

    setTimeout(() => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  },50)
});

it('Update entry', (done) => {
  let data = [
    {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
  ]

  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <Entries reportLink="/report" myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout(() => {
    let dataAfterUpdate = [
      {"id":43,"time_in_second":500,"distance_in_metre":500.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
    ]

    expect(component).toMatchSnapshot()
    stubApiSuccess("get", dataAfterUpdate)
    stubApiSuccess("put", null)
    store.dispatch(updateEntry(1,42, {}))

    setTimeout(() => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  },50)
});

it('Create entry', (done) => {
  let data = [
    {"id":42,"time_in_second":100,"distance_in_metre":100.0,"formatted_date":"2017-09-03","speed":1.0,"date":"2017-09-03"},
    {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"}
  ]

  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <Entries reportLink="/report" myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout(() => {
    let dataAfterUpdate = [
      {"id":42,"time_in_second":100,"distance_in_metre":100.0,"formatted_date":"2017-09-03","speed":1.0,"date":"2017-09-03"},
      {"id":43,"time_in_second":200,"distance_in_metre":200.0,"formatted_date":"2017-09-06","speed":1.0,"date":"2017-09-06"},
      {"id":44,"time_in_second":300,"distance_in_metre":300.0,"formatted_date":"2017-09-09","speed":1.0,"date":"2017-09-09"}
    ]

    expect(component).toMatchSnapshot()
    stubApiSuccess("get", dataAfterUpdate)
    stubApiSuccess("post", null)
    store.dispatch(createEntry(1,{}))

    setTimeout(() => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  },50)
});
