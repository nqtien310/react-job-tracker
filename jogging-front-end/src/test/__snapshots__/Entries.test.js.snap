// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create entry 1`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Sunday, September 3rd 2017
        </td>
        <td>
          100
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Create entry 2`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Sunday, September 3rd 2017
        </td>
        <td>
          100
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td
          className="date"
        >
          Saturday, September 9th 2017
        </td>
        <td>
          300
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Delete entry 1`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Sunday, September 3rd 2017
        </td>
        <td>
          100
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Delete entry 2`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Listing (No Entries) 1`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <div>
    No Entries created
  </div>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Listing (With entries) 1`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Sunday, September 3rd 2017
        </td>
        <td>
          100
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Update entry 1`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          200
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Update entry 2`] = `
<div>
  <div
    className="date-filters"
  >
    Filters:
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="From.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="form-control date-filter"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="To.."
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Distance
        </th>
        <th>
          Time
        </th>
        <th>
          Speed
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="date"
        >
          Wednesday, September 6th 2017
        </td>
        <td>
          500
           m
        </td>
        <td />
        <td>
          1
           m/s
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    href="/report"
    onClick={[Function]}
  >
    Report by week
  </a>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;
